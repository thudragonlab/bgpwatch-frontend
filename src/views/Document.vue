<template>
  <div>
    <el-button class="download-video" @click="downloadVideo">Video for BGPWatch</el-button>
    <iframe :src="filepath" style="height: 90vh; width: 100%" />
  </div>
</template>

<script>
export default {
  name: 'Test',
  created() {},
  data() {
    return {
      filepath: 'https://www.cgtf.net/wp-content/uploads/2023/07/User-Manual-for-BGPWatch.pdf',
      // filepath: 'https://www.bgper.net/wp-content/uploads/2023/03/User-Manual-for-BGPWatch-202303version.pdf',
      
    }
  },
  props: {},
  methods: {
    downloadVideo() {
      window.open('https://www.cgtf.net/wp-content/uploads/2023/07/BGP-Watch-Video-V4.mp4', '_blank')
      // window.open('https://drive.google.com/file/d/1RW-HofbbFKU4LM_xET-uC82r9b83bt34/view', '_blank')
      
    },
  },
}
</script>

<style lang="scss" scoped>
.download-video {
  width: 100%;
  background-color: var(--bg-linear-unclick-start-color);
  color: #fff;
  border-width: 0px;
  font-weight: bold;
  border-radius: 0px;
}
</style>
