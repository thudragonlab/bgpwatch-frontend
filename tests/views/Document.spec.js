import {
  mount
} from '@vue/test-utils'
import Document from "@/views/Document";
import ElementUI from '@/lib/element.js'
import Vue from "vue";

window.open = jest.fn()
Vue.use(ElementUI)

describe('Document.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(Document, {
      stubs: {
        transition: false
      }
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should call gotoDocument with the correct arguments', () => {
    const docBtn = wrapper.find('.download-video')
    docBtn.trigger('click')
    // expect(window.open).toHaveBeenCalledWith('https://www.cgtf.net/wp-content/uploads/2023/07/BGP-Watch-Video-V4.mp4', '_blank')
    expect(window.open).toHaveBeenCalledWith('https://www.cgtf.net/wp-content/uploads/2023/07/BGP-Watch-Video-V4.mp4', '_blank')
    
  });
})