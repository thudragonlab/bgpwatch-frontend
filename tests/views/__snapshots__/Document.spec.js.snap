// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document.vue renders correctly 1`] = `
<div><button type="button" class="el-button download-video el-button--default">
    <!---->
    <!----><span>Video for BGPWatch</span></button> <iframe src="https://www.cgtf.net/wp-content/uploads/2023/07/User-Manual-for-BGPWatch.pdf" style="height: 90vh; width: 100%;"></iframe></div>
`;
